Mailer/app/api/v1/Handlers/Mailers/MailLogsHandler.php

```diff
@@ -2,54 +2,55 @@
 
 namespace Remp\MailerModule\Api\v1\Handlers\Mailers;
 
+use Nette\Utils\Arrays;
 use Nette\Utils\Json;
-use Nette\Utils\JsonException;
-use Remp\MailerModule\Repository\ListsRepository;
+use Remp\MailerModule\ActiveRow;
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
 use Remp\MailerModule\Repository\LogsRepository;
-use Remp\MailerModule\Repository\MailTypesRepository;
 use Tomaj\NetteApi\Handlers\BaseHandler;
-use Tomaj\NetteApi\Params\InputParam;
 use Tomaj\NetteApi\Response\JsonApiResponse;
-use Tracy\Debugger;
 
 class MailLogsHandler extends BaseHandler
 {
     private $logsRepository;
 
+    use InputValidatorTrait;
+
     public function __construct(
         LogsRepository $logsRepository
     ) {
         parent::__construct();
         $this->logsRepository = $logsRepository;
     }
 
-    public function params()
-    {
-        return [
-            new InputParam(InputParam::TYPE_POST, 'where', InputParam::OPTIONAL),
-            new InputParam(InputParam::TYPE_POST, 'limit', InputParam::OPTIONAL),
-        ];
-    }
-
-    public function handle($params)
+    public function handle($payload)
     {
-        $output = $this->logsRepository->all();
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'logs.schema.json');
 
-        if (!empty($params['where'])){
-            $output->where(Json::decode($params['where'], Json::FORCE_ARRAY));
+        if ($this->hasErrors()){
+            return $payload;
         }
 
-        if (!empty($params['limit'])){
+        $output = $this->logsRepository->all()->where($payload['filter'] . ' IS NOT NULL');
 
-            $params['limit'] = Json::decode($params['limit'], Json::FORCE_ARRAY);
+        if (!empty($payload['email'])){
+            $output->where('email', $payload['email']);
+        }
+
+        if (!empty($payload['mail_template_ids'])){
+            $output->where('mail_template_id', $payload['mail_template_id']);
+        }
 
-            if (is_array($params['limit'])){
-                $output->limit($params['limit'][0], $params['limit'][1]);
-            } else {
-                $output->limit($params['limit']);
-            }
+        if (!empty($payload['limit'])){
+            $payload['limit'] = Json::decode($payload['limit'], Json::FORCE_ARRAY);
+            $output->limit($payload['limit'][0], $payload['limit'][1]);
```

Comment:
Why would you make limit two-items array, when you can actually ask in schema for limit AND page?